src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/AutomatiskMigreringService.kt

```diff
@@ -2,40 +2,55 @@ package no.nav.familie.ef.sak.behandling.migrering
 
 import no.nav.familie.ef.sak.infotrygd.InfotrygdReplikaClient
 import no.nav.familie.ef.sak.opplysninger.personopplysninger.secureLogger
+import no.nav.familie.ef.sak.repository.findByIdOrThrow
+import no.nav.familie.log.mdc.MDCConstants
+import no.nav.familie.prosessering.domene.TaskRepository
+import org.jboss.logging.MDC
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
 
 @Service
 class AutomatiskMigreringService(private val migreringsstatusRepository: MigreringsstatusRepository,
                                  private val migreringService: MigreringService,
-                                 private val infotrygdReplikaClient: InfotrygdReplikaClient) {
+                                 private val infotrygdReplikaClient: InfotrygdReplikaClient,
+                                 private val taskRepository: TaskRepository) {
 
     private val logger: Logger = LoggerFactory.getLogger(javaClass)
 
-    // Denne skal ikke gjøres i en transaksjon, då hver migrering håndteres hver for seg
+    @Transactional
     fun migrerAutomatisk(antall: Int) {
         val personerForMigrering = infotrygdReplikaClient.hentPersonerForMigrering(500)
         val alleredeMigrert = migreringsstatusRepository.findAllByIdentIn(personerForMigrering).map { it.ident }
+
         val filtrerteIdenter = personerForMigrering.filterNot { alleredeMigrert.contains(it) }
                 .take(antall) // henter fler fra infotrygd enn vi skal migrere, men plukker ut første X antall
-        logger.info("Automatisk migrering - antall=$filtrerteIdenter")
-        filtrerteIdenter.forEach { migrerPerson(it) }
-        logger.info("Automatisk migrering utført")
+
+        logger.info("Oppretter task for å migrere ${filtrerteIdenter.size} personer")
+        migreringsstatusRepository.insertAll(filtrerteIdenter.map { Migreringsstatus(it, MigreringResultat.IKKE_KONTROLLERT) })
+        taskRepository.save(AutomatiskMigreringTask.opprettTask(filtrerteIdenter.toSet()))
     }
 
-    private fun migrerPerson(personIdent: String) {
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    fun migrerPersonAutomatisk(personIdent: String) {
+        val migreringStatus = migreringsstatusRepository.findByIdOrThrow(personIdent)
+        if (migreringStatus.status != MigreringResultat.IKKE_KONTROLLERT) return
         try {
-            secureLogger.info("Automatisk migrering av ident=$personIdent")
+            val taskCallId = MDC.get(MDCConstants.MDC_CALL_ID)
+            val callId = UUID.randomUUID()
+            // setter nytt callId, sånn att alle nye tasker ikke har samme callId som batch-migrerings-tasken
```

Comment:
Er det fordi det blir "for mange" vi ønsker å splitte på callId? Nå vil det jo bli én og én rad i prosessering iallefall - kan bli veldig mange sider med migrering og lite annet da? 🤔

Reply:
Diskusjon; Lager én task per person for å se flyten mellom migreringstask og iverksettingstask. Samt mulighet for å rekjøre ved behov.